Feature/detail fragment

benjivou opened the pull request.

benjivou committed:
+ cleanup

benjivou committed:
+ cleanup full

benjivou committed:
+ navigation from 1 element to the other

benjivou committed:
+ Refactoring
+ add the id to the navgraph

benjivou committed:
+ add the subMenu

benjivou committed:
+ onglet refactoring

benjivou committed:
+ ignore liked list

benjivou committed:
Merge branch 'develop' into feature/sortingLikes

benjivou committed:
+ activity fragment done

benjivou committed:
Merge branch 'feature/sortingLikes' into feature/detailFragment

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/example/movieapp/MainActivity.kt
#	app/src/main/java/com/example/movieapp/MainFragment.kt
#	app/src/main/java/com/example/movieapp/crawler/ResultPage.kt
#	app/src/main/java/com/example/movieapp/crawler/pojo/ResultPage.kt
#	app/src/main/java/com/example/movieapp/data/dao/AppDatabase.kt
#	app/src/main/java/com/example/movieapp/data/dao/MoviesDAO.kt
#	app/src/main/java/com/example/movieapp/data/model/ResultPage.kt
#	app/src/main/java/com/example/movieapp/model/Movie.kt
#	app/src/main/java/com/example/movieapp/model/ResultPage.kt
#	app/src/main/java/com/example/movieapp/ui/activity/MainActivity.kt
#	app/src/main/java/com/example/movieapp/ui/fragment/MainFragment.kt
#	app/src/main/java/com/example/movieapp/view/MainActivity.kt
#	app/src/main/java/com/example/movieapp/view/fragment/MainFragment.kt
#	app/src/main/java/com/example/movieapp/view/fragment/holder/ListAdapter.kt
#	app/src/main/java/com/example/movieapp/view/fragment/holder/MovieViewHolder.kt
#	app/src/main/java/com/example/movieapp/viewModel/MainViewModel.kt
#	app/src/main/res/layout/activity_fragment.xml
#	app/src/main/res/navigation/nav_graph.xml

benjivou committed:
no message

benjivou committed:
+ Mock navigation

benjivou committed:
+ id past as args

benjivou committed:
+ Add details

benjivou committed:
+ modif gitignore

benjivou committed:
+ gitignore

benjivou committed:
+gitignore

benjivou committed:
no message

benjivou committed:
+ Like connected to Detail View

benjivou committed:
Merge branch 'develop' into feature/detailFragment

# Conflicts:
#	app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt

benjivou committed:
+ front modifications

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt:
```diff
@@ -51,11 +55,17 @@ class MainViewModel : ViewModel() {
 
     val moviesViewHolderListener = object : MovieViewHolder.MoviesViewHolderListener {
```
The viewModel should not do this. You have to make your fragment implement the ViewHolder interface the make the actions from the callback in your fragment

benjivou replied:
dasn le fragment implémenter le view holder. adapter passe this

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt:
```diff
@@ -51,11 +55,17 @@ class MainViewModel : ViewModel() {
+        ) {
+            Log.d(TAG, "onDetailsRequested: image is clicked")
+            val action =
+                MainFragmentDirections.actionMainFragmentToDetailFragment(movie.id.toString())
```
Navigation should be done from the fragment, plus, why do you convert your int into a string then in your DetailViewModel convert your string into a int 🧐

alba-benoit commented on app/src/main/res/layout/detail_fragment.xml:
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
It is best to use a ConstraintLayout, btw the layout is not very sexy =D

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt:
```diff
@@ -0,0 +1,75 @@
+    private var currentId: MutableLiveData<String?> = MutableLiveData()
+    private var currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
+    private var movieCurrent = Transformations.switchMap(currentId) {
+        it?.let { internetCall(it) } ?: null
```
?: null is useless

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt:
```diff
@@ -0,0 +1,75 @@
+
+    init {
+        currentMoviePair.addSource(movieCurrent) {
+            currentMoviePair.value = Pair(it, isLikedMovie?.value ?: false)
```
isLikedMovie is not nullable

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    fun getMovieAndIsLiked(idMovie: String): LiveData<Pair<Movie, Boolean>> {
```
why not returning a MediatorLiveData<Pair<Movie?, Boolean>> ?

benjivou replied:
Return null 
L'abonnement se fait dans une autre méthode ( _currentMoviePaire)

benjivou replied:
private var _currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
val currentMoviePair: LiveData<Pair<Movie?, Boolean>>
    get() = _currentMoviePair

benjivou committed:
+ Move interface implementation to the fragment instead of the ViewModel

benjivou committed:
- AbstractViewModel
+ navigation send Int instead of String

benjivou committed:
+ use Constraint Layout

benjivou committed:
+ correction of the issue "crash when rotate on detailFragment"
+ change the return way of the currentMoviePair

benjivou committed:
+ Cleanup

benjivou committed:
+ creation of the MoviePrepared sealed class

benjivou committed:
Detail viewModel handler internet call errors

benjivou committed:
+ The mainFragment use MoviePrepared
+ Add the title of the list of Movies displayed

benjivou committed:
+ Center title

benjivou committed:
+ Change name of the lists' title
+ convert mainfragment to Constraint Layout
+ Rename Xml values to CamelCase type

benjivou committed:
+ add Padding

benjivou committed:
+ Correct MoviePrepared and Correct MainViewModel

benjivou committed:
- Remove casts
+ Title always links to the Activity
+ Change MoviePrepared in all classes

benjivou committed:
+ Like manager on MainFragment work now

benjivou committed:
+ refactoring code

benjivou committed:
+ Refactoring

benjivou committed:
+Visuel Model change for tablets

alba-benoit commented on app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt:
```diff
@@ -0,0 +1,90 @@
+        _currentMoviePair.addSource(isLikedMovie)
+        { isLiked ->
+            Log.d(TAG, "isLikedMLovie : Modified")
+            val toto = _currentMoviePair.value
```
who is toto ?

benjivou committed:
+ rename variablee

benjivou committed:
refactoring
